### PR TITLE
Modules config

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/Modules/SimplCommerce.Module.Core/Views/HomeAdmin/Index.cshtml [hunk 1]
```diff
@@ -151,9 +151,9 @@
     <script simpl-append-version="true" src="~/modules/comments/admin/comment/comment-service.js"></script>
     <script simpl-append-version="true" src="~/modules/comments/admin/comment/comment-list.js"></script>
 
-    <script simpl-append-version="true" src="~/modules/activitylog/admin/activitylog.module.js"></script>
-    <script simpl-append-version="true" src="~/modules/activitylog/admin/activitylog-service.js"></script>
-    <script simpl-append-version="true" src="~/modules/activitylog/admin/most-viewed-products.directive.js"></script>
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Core/Views/HomeAdmin/Index.cshtml]: activitylog is the correct one

File: src/Modules/SimplCommerce.Module.Core/Views/HomeAdmin/Index.cshtml [hunk 2]
```diff
@@ -200,9 +200,9 @@
     <script simpl-append-version="true" src="~/modules/shipping/admin/provider/shipping-provider-service.js"></script>
     <script simpl-append-version="true" src="~/modules/shipping/admin/provider/shipping-provider-list.js"></script>
 
-    <script simpl-append-version="true" src="~/modules/shippingtablerate/admin/shipping-tablerate.module.js"></script>
-    <script simpl-append-version="true" src="~/modules/shippingtablerate/admin/tablerate/shipping-tablerate-service.js"></script>
-    <script simpl-append-version="true" src="~/modules/shippingtablerate/admin/tablerate/shipping-tablerate-form.js"></script>
+    <script simpl-append-version="true" src="~/modules/shipping-tablerate/admin/shipping-tablerate.module.js"></script>
+    <script simpl-append-version="true" src="~/modules/shipping-tablerate/admin/tablerate/shipping-tablerate-service.js"></script>
+    <script simpl-append-version="true" src="~/modules/shipping-tablerate/admin/tablerate/shipping-tablerate-form.js"></script>
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Core/Views/HomeAdmin/Index.cshtml]: shippingtablerate is the correct one

File: src/Modules/SimplCommerce.Module.Core/Views/HomeAdmin/Index.cshtml [hunk 3]
```diff
@@ -222,9 +222,9 @@
     <script simpl-append-version="true" src="~/modules/paymentcod/admin/config/cod-service.js"></script>
     <script simpl-append-version="true" src="~/modules/paymentcod/admin/config/config-form.js"></script>
 
-    <script simpl-append-version="true" src="~/modules/productrecentlyviewed/admin/product-recently-viewed.module.js"></script>
-    <script simpl-append-version="true" src="~/modules/productrecentlyviewed/admin/recently-viewed-widget/recently-viewed-widget-service.js"></script>
-    <script simpl-append-version="true" src="~/modules/productrecentlyviewed/admin/recently-viewed-widget/recently-viewed-widget-form.js"></script>
+    <script simpl-append-version="true" src="~/modules/product-recently-viewed/admin/product-recently-viewed.module.js"></script>
+    <script simpl-append-version="true" src="~/modules/product-recently-viewed/admin/recently-viewed-widget/recently-viewed-widget-service.js"></script>
+    <script simpl-append-version="true" src="~/modules/product-recently-viewed/admin/recently-viewed-widget/recently-viewed-widget-form.js"></script>
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Core/Views/HomeAdmin/Index.cshtml]: productrecentlyviewed is the correct one

**Comment by hishamco** [comment]
Sure

**Comment by thiennn** [comment]
Honestly, I don't see a clear benefits of adding modules.json to keep track installed modules instead of scanning Modules folder. Maintain modules.json will add more task for developers to take care.

**Comment by hishamco** [comment]
> Honestly, I don't see a clear benefits of adding modules.json to keep track installed modules instead of scanning Modules folder

We can avoid using Modules folder which is contains all the modules that copied by gulp tasks, think of this as NuGet packages, each package has it's own folder and all the installed packages are written in `packages.config`

**Comment by hishamco** [comment]
@thiennn if you don't like the NuGet like style, we can go with another way: modules & themes will remain same as what you did before, but what me as developer should expect is all the modules inside that folder are the installed once, similarly on the themes, nothing but if we consider the dependencies this should go during the installation of a new module to validate if the dependencies, similar to what I did in my WIP PR #581

**Comment by hishamco** [comment]
@thiennn please have a look to my previous comment to finalize how we can end up with this PR

File: src/Modules/SimplCommerce.Module.Core/Views/HomeAdmin/Index.cshtml (hunk 1 above)
Comment (thiennn) [reply src/Modules/SimplCommerce.Module.Core/Views/HomeAdmin/Index.cshtml]: Please revert back this change

**Comment by thiennn** [comment]
The idea of using modules.json is good. But I think we should use this file to determine which modules are copied to the Host during the gulp-copy-modules. And during the load module, we still scan the Modules folder in the WebHost as current.

After this change, we are able to merge #599

**Comment by hishamco** [comment]
I resolved the conflict changes

**Comment by thiennn** [comment]
We cannot target to netcoreapp2.1 on the SimplCommerce.MSBuildTasks. Because visual studio will call MSBuild.exe (.NET Full Framework) to build projects.

There are many issues when loading dependencies of custom and msbuild task. That is the reason I tried not to use any external libraries for the custom build tasks

https://natemcmaster.com/blog/2017/07/05/msbuild-task-in-nuget/

**Comment by hishamco** [comment]
I build the project many times using Visual Studio many times without any issues, BTW this blog post since year and so.
@natemcmaster your advice if you don't mind about last comment

**Comment by thiennn** [comment]
You build without issue because it is still you the old binary.

`<UsingTask AssemblyFile="..\..\build\netstandard2.0\SimplCommerce.MSBuildTasks.dll" TaskName="CopyModuleTask" />`

**Comment by natemcmaster** [comment]
> We cannot target to netcoreapp2.1 on the SimplCommerce.MSBuildTasks. Because visual studio will call MSBuild.exe (.NET Full Framework) to build projects.

@thiennn is right. If you want to write MSBuild tasks which work with both `dotnet build` and Visual Studio, you need to cross-compile your MSBuild tasks for .NET Framework and .NET Core. I recommend tasks assemblies compile like this:
```xml
<TargetFrameworks>netcoreapp2.0;net46</TargetFrameworks>
```
Then, load the right task assembly based on the MSBuild runtime version.
```xml
  <PropertyGroup>
    <TaskFolder Condition=" '$(MSBuildRuntimeType)' == 'Core' ">netcoreapp2.0</TaskFolder>
    <TaskFolder Condition=" '$(MSBuildRuntimeType)' != 'Core' ">net46</TaskFolder>
    <TaskAssembly>$(MSBuildThisFileDirectory)build\$(TaskFolder)\SimplCommerce.MSBuildTasks.dll.dll</TaskAssembly>
  </PropertyGroup>

  <UsingTask TaskName="CopyModuleTask" AssemblyFile="$(TaskAssembly)" />
```
More details about this approach here: https://natemcmaster.com/blog/2017/07/05/msbuild-task-in-nuget/. A working sample here: https://github.com/natemcmaster/msbuild-tasks

**Comment by hishamco** [comment]
I will modify the PR according your suggestion, thanks @natemcmaster

**Comment by hishamco** [comment]
Seems we have an issue there because `SimplCommerce.Infrastructure` is target `netcoreapp2.1`, so if we re-target  `SimplCommerce.MSBuildTasks` to ` netcoreapp2.1` the `SimplCommerce.Infrastructure` will complains because we target `net46` 😞

**Comment by thiennn** [comment]
@natemcmaster I did as what you commented above, it worked. Then I tried to test in with <TargetFramework>netstandard2.0</TargetFramework> and it works to for both Visual Studio and command line. Because netcoreapp and net full framework can consume netstandard 2.0 ?

**Comment by thiennn** [comment]
@hishamco I don't think making SimplCommerce.MSBuildTasks depend on SimplCommerce.Infrastructure is a good idea. I have tried to make SimplCommerce.MSBuildTasks as simple as possible, eliminated all the external dependencies.  There are many problems when a MsBuild task has external dependencies.

**Comment by thiennn** [comment]
Btw, I think it's ok to let the SimplCommerce.MSBuildTasks target to netstandard2.0 as currently. This build task is only used in SimplCommerce, which will required VS 2017 18.x, and we are also not using any special  API in this custom build task

**Comment by hishamco** [comment]
@thiennn I revert the changes ..

**Comment by thiennn** [comment]
To be honest, I still refer scanning all the directories in the Modules folders to get the list of installed module. So that there is only one single source of truth, and it also consistency with the current theme discovery. However, I gonna merge this PR, trying this approach for some time

**Pull request merged**